File: src/main/java/io/stouder/adonis/edge/parsing/EdgeLexer.java
package io.stouder.adonis.edge.parsing;

import com.intellij.lexer.MergingLexerAdapter;
import com.intellij.psi.tree.TokenSet;

public class EdgeLexer extends MergingLexerAdapter {
    public EdgeLexer() {
        super(new EdgeRawLexer(), TOKENS_TO_MERGE);
    }

    private final static TokenSet TOKENS_TO_MERGE = TokenSet.create(
            EdgeTokenTypes.CONTENT,
            EdgeTokenTypes.MUSTACHE_CONTENT,
            EdgeTokenTypes.WHITE_SPACE
    );

}

File: src/main/java/io/stouder/adonis/edge/parsing/EdgeLexer.kt
package io.stouder.adonis.edge.parsing

import com.intellij.lexer.MergingLexerAdapter
import com.intellij.psi.tree.TokenSet

class EdgeLexer(tokensToMerge: TokenSet?) : MergingLexerAdapter(EdgeRawLexer(), tokensToMerge) {

    companion object {
        private val TOKENS_TO_MERGE = TokenSet.create(
            EdgeTokenTypes.CONTENT,
            EdgeTokenTypes.MUSTACHE_CONTENT,
            EdgeTokenTypes.COMMENT_MUSTACHE_CONTENT,
            EdgeTokenTypes.TAG_CONTENT,
            EdgeTokenTypes.WHITE_SPACE
        )
    }

    constructor() : this(TOKENS_TO_MERGE)
}
